### PR TITLE
ESS Staging integ test improvements

### PR DESCRIPTION
**Pull request by pchila**
## What does this PR do?

- refactor deployment creation and support aws templates
- support tags to be assigned to integration tests stack deployments

## Why is it important?
Support all 3 major CSP and tag stack deployments appropriately

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Author's Checklist

- [ ]

## Related issues

- 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -1736,7 +1736,7 @@ func createTestRunner(matrix bool, singleTest string, goTestFlags string, batche
 	// Valid values are gcp-us-central1 (default), azure-eastus2
 	essRegion := os.Getenv("TEST_INTEG_AUTH_ESS_REGION")
 	if essRegion == "" {
-		essRegion = "gcp-us-central1"
+		essRegion = "aws-eu-central-1"
```
Comment (cmacknz) [inline magefile.go:1739]: We need to double check that changing cloud providers for deployment provisioning doesn't cause us any unexpected extra expenses.

CC @alexsapran

Comment (pchila) [reply magefile.go:1739]: I am not sure why you would expect extra expenses for GCP <-> AWS traffic as compared to GCP <-> Azure as we did when switching regions in QA...
Is there something specific to GCP <-> AWS data exchange ?

**Comment by ycombinator** [comment]
> There are dev features that are exclusive to the CFT regions

I think I'm missing some background on this PR.  Why are we moving from GCP to AWS in the staging region? From what I can tell both GCP and AWS regions support the same stack versions, including snapshots.  What other dev features are exclusive to the AWS region in staging that aren't available in the GCP region in staging?

**Comment by pchila** [comment]
buildkite test this

File: magefile.go (hunk 1 above)
Comment (pchila) [reply magefile.go]: Restored gcp region as default region. I will leave the aws support in anyways...

**Comment by pchila** [comment]
buildkite test this

**Comment by pchila** [comment]
buildkite test this

**Review by ycombinator — APPROVED** [review]

**Pull request merged**